src/org/omegat/core/data/RealProject.java

```diff
@@ -1141,35 +1179,50 @@ private void loadSourceFiles() throws Exception {
 
-            loadFilesCallback.setCurrentFile(fi);
-
-            IFilter filter = fm.loadFile(config.getSourceRoot() + filepath, new FilterContext(config),
-                    loadFilesCallback);
-
-            loadFilesCallback.fileFinished();
-
-            if (filter != null && !fi.entries.isEmpty()) {
-                fi.filterClass = filter.getClass(); //Don't store the instance, because every file gets an instance and
-                                                    // then we consume a lot of memory for all instances.
-                                                    //See also IFilter "TODO: each filter should be stateless"
-                fi.filterFileFormatName = filter.getFileFormatName();
-                try {
-                    fi.fileEncoding = filter.getInEncodingLastParsedFile();
-                } catch (Error e) { // In case a filter doesn't have getInEncodingLastParsedFile() (e.g., Okapi plugin)
-                    fi.fileEncoding = "";
+            try {
+                loadFilesCallback.setCurrentFile(fi);
+                IFilter filter = fm.loadFile(config.getSourceRoot() + filepath, new FilterContext(config),
+                        loadFilesCallback);
+                loadFilesCallback.fileFinished();
+
+                if (filter != null && !fi.entries.isEmpty()) {
+                    fi.filterClass = filter.getClass(); // Don't store the
+                                                        // instance, because
+                                                        // every file gets an
+                                                        // instance and
+                                                        // then we consume a lot
+                                                        // of memory for all
+                                                        // instances.
+                                                        // See also IFilter
+                                                        // "TODO: each filter
+                                                        // should be stateless"
+                    fi.filterFileFormatName = filter.getFileFormatName();
+                    try {
+                        fi.fileEncoding = filter.getInEncodingLastParsedFile();
+                    } catch (Error e) { // In case a filter doesn't have
+                                        // getInEncodingLastParsedFile() (e.g.,
+                                        // Okapi plugin)
+                        fi.fileEncoding = "";
+                    }
+                    projectFilesList.add(fi);
                 }
-                projectFilesList.add(fi);
+            } catch (TranslationException e) {
+                // ignore failed file and continue loading next.
+                Log.logErrorRB("TF_SOURCE_LOAD_ERROR", e.getLocalizedMessage());
+                Core.getMainWindow().displayErrorRB(e, "TF_SOURCE_LOAD_ERROR", filepath);
             }
```

Comment:
When we got `TranslationException` from `FilterMaster` class, OmegaT notify the error to user in error dialog with problematic filename, then continue the process.

Reply:
It would be better to insert explicit call of `continue` to express we don't abort here.